src/fidesops/ops/service/privacy_request/request_runner_service.py

```diff
@@ -343,20 +354,70 @@ async def run_privacy_request(
         )
         if not proceed:
             return
-
+        if config.notifications.send_request_completion_notification:
+            try:
+                initiate_privacy_request_completion_email(
+                    session, policy, access_result_urls, identity_data
+                )
```

Comment:
We also need to decide what happens if the email doesn't send and we put the PrivacyRequest in an error state (in the EmailDispatchException case), what is the best course of action?  

Do they submit a new PrivacyRequest? (If there are erasure rules defined it is possible that this request can't be re-run)

Do they restart from this point? To be able to restart from failure, you need to cache a checkpoint:  `privacy_request.cache_failed_checkpoint_details(
step=CurrentStep.post_notification_send(new checkpoint), collection=None)` so fidesops knows where to resume the request.  Is there any way to be able to rebuild the S3 link in the case of an access request? For an erasure, we could just attempt to resend just the email.

Reply:
To help answer this, let's talk about 2 likely issues with email send:

1. Something is misconfigured with the user's email server, e.g. proper domain sending validation / MX records haven't been set up.
2. User email domain rejects email for some reason, or isn't valid

I think it makes sense, in case 1, to allow for restarting from failure. We can probably regenerate a new signed URL (which renews the expiry) as long as we know the privacy request ID. I'll add this in.

For case 2, I don't think anything can be done about an invalid email domain, bc we can't email them to submit a new request.

Something I want to add in the future is implementing retries upon failure, regardless of reason.

Reply:
Had a chat with @seanpreston and it's OK to bypass handling these cases for now. Also a note that once his Celery work is merged, we can have Celery handle retries with a simple annotation.

Reply:
OK I'd still catch the error and put it the PrivacyRequest in a failed state, instead of leaving it as in-processing, even if we don't allow this to be retried.